ksql-test-commons/src/test/java/io/confluent/ksql/test/commons/TestCaseTest.java

```diff
@@ -0,0 +1,218 @@
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import io.confluent.ksql.metastore.model.KsqlTopic;
+import io.confluent.ksql.serde.json.KsqlJsonTopicSerDe;
+import io.confluent.ksql.util.KafkaTopicClient;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.streams.TopologyTestDriver;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestCaseTest {
+
+  @Rule
+  public final org.junit.rules.ExpectedException expectedException = org.junit.rules.ExpectedException
+      .none();
+
+  private final SerdeSupplier serdeSupplier = new StringSerdeSupplier();
+  private final Topic topic = new Topic("foo_kafka", Optional.empty(), serdeSupplier, 4, 1);
+  private final Record record = new Record(topic, "k1", "v1, v2", 123456789L, null);
+  private final TestCase testCase = new TestCase(
+      null,
+      "test",
+      Collections.emptyMap(),
+      ImmutableList.of(topic),
+      ImmutableList.of(record),
+      ImmutableList.of(record),
+      Collections.emptyList(),
+      ExpectedException.none(),
+      PostConditions.NONE
+  );
+
+
+  @Mock
+  private TopologyTestDriver topologyTestDriver;
+
+
+  @Before
+  public void init() {
+
+  }
+
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldProcessInputRecords() {
+    // Given:
+    final TopologyTestDriverContainer topologyTestDriverContainer = getSampleTopologyTestDriverContainer();
+
+
+    // When:
+    testCase.processInput(topologyTestDriverContainer, null);
+
+
+    // Then:
+    verify(topologyTestDriver).pipeInput(any(ConsumerRecord.class));
+
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void shouldFilterNonSourceTopics() {
+    // Given:
+    final TopologyTestDriverContainer topologyTestDriverContainer = TopologyTestDriverContainer.of(
+        topologyTestDriver,
+        ImmutableSet.of(new KsqlTopic("FOO", "foo_kafka_different_input", new KsqlJsonTopicSerDe(), false)),
+        ImmutableSet.of(new KsqlTopic("BAR", "bar_kafka", new KsqlJsonTopicSerDe(), false))
+    );
+
+    // When:
+    testCase.processInput(topologyTestDriverContainer, null);
+
+    // Then:
+    verify(topologyTestDriver, never()).pipeInput(any(ConsumerRecord.class));
+
+  }
+
+  @Test
+  public void shouldValidateOutputCorrectly() {
+    // Given:
+    final TopologyTestDriverContainer topologyTestDriverContainer = getSampleTopologyTestDriverContainer();
+    when(topologyTestDriver.readOutput(any(), any(), any()))
+        .thenReturn(new ProducerRecord<>("bar_kafka", 1, 123456789L, "k1", "v1, v2"));
+
+
+    // When:
```

Comment:
```suggestion
    // No Exception When:
```

Reply:
I didn't know we also use `No Exception`! have we used this pattern before?

Reply:
I've seen some code that does it (cc @big-andy-coates) if there is no other `assert` style statement or expected exception to make it clear what this test does

Reply:
I tend to use 

```java
   // Then: no exception
```